### PR TITLE
fix celery error and update celery document

### PR DESCRIPTION
**Pull request by lancemao**
# Description

* Update docuemt to make it clear that in order to use knowledge base, developer MUST start celery locally
* Fix error when starting celery due to missing module

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement，including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/requirements.txt [hunk 1]
```diff
@@ -73,4 +73,5 @@ yarl~=1.9.4
 twilio==9.0.0
 qrcode~=7.4.2
 azure-storage-blob==12.9.0
-azure-identity==1.15.0
\ No newline at end of file
+azure-identity==1.15.0
+tzdata==2024.1
```
Comment (takatost) [inline api/requirements.txt:77]: I think tzdata isn't necessary tho, mind telling me what's the issue?

**Review by takatost — CHANGES_REQUESTED** [review]

File: api/requirements.txt (hunk 1 above)
Comment (lancemao) [reply api/requirements.txt:77]: without it, I got the following error when I try to start celery:

```
(.venv) PS E:\OpenSource\dify\api> celery -A app.celery worker -P gevent -c 1 --loglevel INFO -Q dataset,generation,mail
Traceback (most recent call last):
  File "C:\Users\kaihong\AppData\Local\Programs\Python\Python311\Lib\zoneinfo\_common.py", line 12, in load_tzdata
    return resources.files(package_name).joinpath(resource_name).open("rb")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kaihong\AppData\Local\Programs\Python\Python311\Lib\importlib\resources\_common.py", line 22, in files
    return from_package(get_package(package))
                        ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kaihong\AppData\Local\Programs\Python\Python311\Lib\importlib\resources\_common.py", line 53, in get_package
    resolved = resolve(package)
               ^^^^^^^^^^^^^^^^
  File "C:\Users\kaihong\AppData\Local\Programs\Python\Python311\Lib\importlib\resources\_common.py", line 44, in resolve
    return cand if isinstance(cand, types.ModuleType) else importlib.import_module(cand)
                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kaihong\AppData\Local\Programs\Python\Python311\Lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'tzdata'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "E:\OpenSource\dify\api\.venv\Scripts\celery.exe\__main__.py", line 7, in <module>
  File "E:\OpenSource\dify\api\.venv\Lib\site-packages\celery\__main__.py", line 13, in main
    maybe_patch_concurrency()
  File "E:\OpenSource\dify\api\.venv\Lib\site-packages\celery\__init__.py", line 146, in maybe_patch_concurrency
    concurrency.get_implementation(pool)
  File "E:\OpenSource\dify\api\.venv\Lib\site-packages\celery\concurrency\__init__.py", line 28, in get_implementation
    return symbol_by_name(cls, ALIASES)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\OpenSource\dify\api\.venv\Lib\site-packages\kombu\utils\imports.py", line 59, in symbol_by_name
    module = imp(module_name, package=package, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kaihong\AppData\Local\Programs\Python\Python311\Lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "E:\OpenSource\dify\api\.venv\Lib\site-packages\celery\concurrency\gevent.py", line 4, in <module>
    from kombu.asynchronous import timer as _timer
  File "E:\OpenSource\dify\api\.venv\Lib\site-packages\kombu\asynchronous\__init__.py", line 7, in <module>
    from .hub import Hub, get_event_loop, set_event_loop
  File "E:\OpenSource\dify\api\.venv\Lib\site-packages\kombu\asynchronous\hub.py", line 20, in <module>
    from .timer import Timer
  File "E:\OpenSource\dify\api\.venv\Lib\site-packages\kombu\asynchronous\timer.py", line 32, in <module>
    EPOCH = datetime.utcfromtimestamp(0).replace(tzinfo=ZoneInfo("UTC"))
                                                        ^^^^^^^^^^^^^^^
  File "C:\Users\kaihong\AppData\Local\Programs\Python\Python311\Lib\zoneinfo\_common.py", line 24, in load_tzdata
    raise ZoneInfoNotFoundError(f"No time zone found with key {key}")
zoneinfo._common.ZoneInfoNotFoundError: 'No time zone found with key UTC'
```

then I installed tzdata, everything is working fine:

```
(.venv) PS E:\OpenSource\dify\api> pip install tzdata
Collecting tzdata
  Obtaining dependency information for tzdata from https://files.pythonhosted.org/packages/65/58/f9c9e6be752e9fcb8b6a0ee9fb87e6e7a1f6bcab2cdc73f02bb7ba91ada0/tzdata-2024.1-py2.py3-none-any.whl.metadata
  Downloading tzdata-2024.1-py2.py3-none-any.whl.metadata (1.4 kB)
Downloading tzdata-2024.1-py2.py3-none-any.whl (345 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 345.4/345.4 kB 3.6 MB/s eta 0:00:00
Installing collected packages: tzdata
Successfully installed tzdata-2024.1

[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip
(.venv) PS E:\OpenSource\dify\api> celery -A app.celery worker -P gevent -c 1 --loglevel INFO -Q dataset,generation,mail
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
E:\OpenSource\dify\api\.venv\Lib\site-packages\pydub\utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
  warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
 
 -------------- celery@OSS-maolongdong-000821-B v5.2.7 (dawn-chorus)
--- ***** -----
-- ******* ---- Windows-10-10.0.22631-SP0 2024-03-21 01:50:36
- *** --- * ---
- ** ---------- [config]
- ** ---------- .> app:         app:0x262255f18d0
- ** ---------- .> transport:   redis://:**@localhost:6379/1
- ** ---------- .> results:     postgresql://postgres:**@localhost:5432/dify
- *** --- * --- .> concurrency: 1 (gevent)
-- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
--- ***** -----
 -------------- [queues]
                .> dataset          exchange=dataset(direct) key=dataset
                .> generation       exchange=generation(direct) key=generation
                .> mail             exchange=mail(direct) key=mail

[tasks]
  . schedule.clean_embedding_cache_task.clean_embedding_cache_task
  . schedule.clean_unused_datasets_task.clean_unused_datasets_task
  . tasks.add_document_to_index_task.add_document_to_index_task
  . tasks.annotation.add_annotation_to_index_task.add_annotation_to_index_task
  . tasks.annotation.batch_import_annotations_task.batch_import_annotations_task
  . tasks.annotation.delete_annotation_index_task.delete_annotation_index_task
  . tasks.annotation.disable_annotation_reply_task.disable_annotation_reply_task
  . tasks.annotation.enable_annotation_reply_task.enable_annotation_reply_task
  . tasks.annotation.update_annotation_to_index_task.update_annotation_to_index_task
  . tasks.batch_create_segment_to_index_task.batch_create_segment_to_index_task
  . tasks.clean_dataset_task.clean_dataset_task
  . tasks.clean_document_task.clean_document_task
  . tasks.clean_notion_document_task.clean_notion_document_task
  . tasks.deal_dataset_vector_index_task.deal_dataset_vector_index_task
  . tasks.delete_segment_from_index_task.delete_segment_from_index_task
  . tasks.disable_segment_from_index_task.disable_segment_from_index_task
  . tasks.document_indexing_sync_task.document_indexing_sync_task
  . tasks.document_indexing_task.document_indexing_task
  . tasks.document_indexing_update_task.document_indexing_update_task
  . tasks.enable_segment_to_index_task.enable_segment_to_index_task
  . tasks.mail_invite_member_task.send_invite_member_mail_task
  . tasks.recover_document_indexing_task.recover_document_indexing_task
  . tasks.remove_document_from_index_task.remove_document_from_index_task

[2024-03-21 01:50:38,637: INFO/MainProcess] Connected to redis://:**@localhost:6379/1
[2024-03-21 01:50:40,682: INFO/MainProcess] mingle: searching for neighbors
[2024-03-21 01:50:47,800: INFO/MainProcess] mingle: all alone
[2024-03-21 01:50:49,860: INFO/MainProcess] pidbox: Connected to redis://:**@localhost:6379/1.
[2024-03-21 01:50:53,959: INFO/MainProcess] celery@OSS-maolongdong-000821-B ready.
```

Comment (bowenliang123) [reply api/requirements.txt:77]: It's not usual to see updates in dependencies while only docs touched in the PR.
The existing code should work well without it.

Comment (bowenliang123) [reply api/requirements.txt:77]: Could you just separate the changes into two, leaving only the doc changes in this PR?

**Pull request closed without merging**